subsys/bluetooth/host/id.c

```diff
@@ -240,6 +240,22 @@ static void le_rpa_invalidate(void)
 
 #if defined(CONFIG_BT_PRIVACY)
 
+struct bt_adv_id_rpa_data {
+	uint8_t id;
+	bool rpa_created_for_adv_id;
+	bt_addr_t *rpa;
```

Comment:
Instead of having the `bool` variable, could you just pre-initialize `rpa` to `BT_ADDR_LE_NONE` and then if it's equal to anything else after the iteration you know that an existing RPA was found?

Reply:
The `rpa` is a pointer, so is that possible? 

Alternatively, could `rpa == NULL` indicate the same?